AsyncDisplayKit/Private/NSArray+Diffing.m

```diff
@@ -0,0 +1,78 @@
+//
+//  NSArray+Diffing.m
+//  AsyncDisplayKit
+//
+//  Created by Levi McCallum on 1/29/16.
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#import "NSArray+Diffing.h"
+
+@implementation NSArray (Diffing)
+
+- (void)asdk_diffWithArray:(NSArray *)array insertions:(NSIndexSet **)insertions deletions:(NSIndexSet **)deletions
+{
+  [self asdk_diffWithArray:array insertions:insertions deletions:deletions compareBlock:^BOOL(id lhs, id rhs) {
+    return [lhs isEqual:rhs];
+  }];
+}
+
+- (void)asdk_diffWithArray:(NSArray *)array insertions:(NSIndexSet **)insertions deletions:(NSIndexSet **)deletions compareBlock:(BOOL (^)(id lhs, id rhs))comparison
```

Comment:
Very cool.  Great, clear code.  Profiling will determine if we need further optimization, but this is probably about as far as we can get without C++.